six/projects/csm/source/SIXSensorModel.cpp

```diff
@@ -1110,5 +1110,62 @@ double SIXSensorModel::getCorrelationCoefficient(size_t cpGroupIndex,
                          "SIXSensorModel::getCorrelationCoefficient");
     }
 }
+
+DataType SIXSensorModel::getDataType(const csm::Des& des)
+{
+    // NOTE: Versions of SICD <= 1.1 and SIDD <= 1.0 prefixed FTITLE with
+    //       SICD or SIDD, so for old files we could key off of that.  Since
+    //       that's not guaranteed for newer versions though, we now use the
+    //       DESID for really old versions and the DESSHSI from
+    //       XML_DATA_CONTENT for newer versions.
+    std::string desid = des.subHeader().substr(NITF_DE_SZ, NITF_DESTAG_SZ);
+    str::trim(desid);
+    // SICD/SIDD 1.0 specify DESID as XML_DATA_CONTENT
+    // Older versions of the spec specified it as SICD_XML/SIDD_XML
+    // Here we'll accept any of these under the assumption that it's not
+    // such an old version of the spec that the XML layout itself has
+    // changed (if it did, XMLControl will end up throwing anyway)
+    if (desid == "SICD_XML")
+    {
+        return DataType::COMPLEX;
+    }
+    else if (desid == "SIDD_XML")
+    {
+        return DataType::DERIVED;
+    }
+    else if (desid == Constants::DES_USER_DEFINED_SUBHEADER_TAG)
+    {
+        // Check whether DES is SICD/SIDD, or something of a different
+        // sort. Note that we need to check the subheader
+        // length first rather than calling tre.getCurrentSize() in
+        // case there is no subheader (in which case
+        // subheader.getSubheaderFields() will throw with a
+        // NITRO-specific message).
+        const std::string& subheader = des.subHeader();
```

Comment:
I’ve made the edits on this, fixTests, and currentPath in CODA-OSS.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Sunday, July 17, 2016 4:24 PM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [ngageoint/six-library] Fix how CSM figures out an images type, to match how NITFReadControl now does it. (#90)


In six/projects/csm/source/SIXSensorModel.cpp<https://github.com/ngageoint/six-library/pull/90#discussion_r71088741>:

> +    {

> +        return DataType::COMPLEX;

> +    }

> +    else if (desid == "SIDD_XML")

> +    {

> +        return DataType::DERIVED;

> +    }

> +    else if (desid == Constants::DES_USER_DEFINED_SUBHEADER_TAG)

> +    {

> +        // Check whether DES is SICD/SIDD, or something of a different

> +        // sort. Note that we need to check the subheader

> +        // length first rather than calling tre.getCurrentSize() in

> +        // case there is no subheader (in which case

> +        // subheader.getSubheaderFields() will throw with a

> +        // NITRO-specific message).

> +        const std::string& subheader = des.subHeader();

XML_DATA_CONTENT is statically linked in now so this can use that now. But can't you take it a step further and use a method from NITFReadControl so that the logic is in one spot (might need to refactor it a bit)?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ngageoint/six-library/pull/90/files/a916174e4a146c08edd518b9bcdcd6491686adc3#r71088741>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFRAYLgI1-SyMSO7ZJ3g4F2BJCxGK-y-ks5qWo9LgaJpZM4JNnoK>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.